Document resetting Rock-on Root (#434) (#503)
rockstor/rockstor-doc: interface/overview.rst

    within each rock-on but that is also good practice, and is why this option
    is offered.

It is assumed that :ref:`pools` and one or more shares in those pools
(see our :ref:`createshare`) have already been created, appropriate for the
Rock-on requirements, e.g. a *plex-movies* share and a *plex-config* share.

> As a one-time setup activity, it is also necessary create

Reviewer: ```suggestion
As a one-time setup activity, it is also necessary to create
```

enabling this service it must be configured. This is a simple matter of
configuring a sufficiently large share for the rock-ons to be installed into.

It is recommended to create a *dedicated* share to be used as the Rock-ons root.
While any share can be chosen as the Rock-ons root, it is **strongly** recommended
to create a share to be used **only** as the Rock-ons root to prevent any

> conflict that may arise with a mixed-used share. The Rock-ons root share is

Reviewer: ```suggestion
conflict that may arise with a mixed-use share. The Rock-ons root share is
```

@Hooverdan96 I know this was the case before, but just noting while I'm here.

used by Docker to store permanent data such as images and container
layers. As a result, any conflict or alteration of these data that might result

> from a mixed used share has the potential to break the installed Rock-ons.

Reviewer: ```suggestion
from a mixed-use share has the potential to break the installed Rock-ons.
```

Again this does not pertain to your alterations but we could slip this in as we go.

*This example is for newer Rockstor versions.*

.. _rockons_root_reset:

> Reset Rock-on Root

Reviewer: ```suggestion
Reset The Rock-ons Root
```

@Hooverdan96 I prefer this reference as it is the root for all rock-ons, not for a single rock-on. And our pics and other doc references hopefully mostly conform to this arbitrary but important reference. [EDIT] ~~I know a 'the' could be nice but we are titles here so I think this floats :) .~~

  and a subsequent attempt to re-install it does not seem to work.
* The Rock-on fails to start when it worked in previous cases without any
  issues. However, this could also be related to a breaking update of
  the underlying image used for the Rock-on.
* The WebUI throws an error like this when uninstalling/reinstalling a
  Rock-on:

  * :code:`Unknown internal error doing a GET to /api/rockons?page...`.

With the symptoms described above, the best resolution might be to reset the Rock-on root,
following the sequence below:

* If possible, note the settings of the still installed Rock-ons.

.. note::

    If, alternatively, using the :ref:`config_backup` option to back-up the 
    settings of installed Rock-ons, then you should follow the
    *recreate* Rock-ons root option described below.

* Uninstall all installed Rock-ons (likely requires the command line as described 
  in :ref:`rockons_force_delete`).

.. note::

   All shares that were used for any of the Rock-ons will not be affected by these process
   steps, so a later re-installation of an associated Rock-on should put the situation

>    right back to where it was before the Rock-on root reset process.

Reviewer: ```suggestion
   right back to where it was before the Rock-ons root reset process.
```

> * Turn off Rock-ons Service (either on the Rockons page directly, or via the 

Reviewer: ```suggestion
* Turn off the Rock-ons Service (either on the Rockons page directly, or via the 
```

  **System --> Services** web page).

> * Following :ref:`rockons_root`, either create a new Rock-on root share with

Reviewer: ```suggestion
* Following :ref:`rockons_root`, to either create a new Rock-ons root share with
```

>   a different name, or delete and recreate the Rock-on root share with its original name.

Reviewer: ```suggestion
  a different name, or delete and recreate the Rock-ons root share with its original name.
```

.. note::

>   In case of creating a new Rock-on root share with a different name, adjust the Rock-ons

Reviewer: ```suggestion
  In the case of creating a new Rock-ons root share with a different name, adjust the Rock-ons
```

>   Service setting by selecting the new share name from the dropdown. If the Rock-on root

Reviewer: ```suggestion
  Service setting by selecting the new share name from the dropdown. If the Rock-ons root
```

  share has been re-created with the same name, no change is necessary. Delete the old
  Rock-on root share to free up disk space.

* Restart the Rock-ons Service.
* Run an update on the list of available Rock-ons.

Now the Rock-ons root should be consistent, albeit empty, once again, and (re-)installation
of Rock-ons should work as before. Or, if the configuration backup was created
this could now be executed, which should skip all existing configuration elements

> and only result in the re-installation of the previously capture Rock-ons.

Reviewer: ```suggestion
and only result in the re-installation of the previously installed Rock-ons.
```
